plover/gui/main.py

```diff
@@ -96,80 +102,112 @@ def __init__(self, config):
                                                bitmap=self.disconnected_bitmap)
 
         # Layout.
-        global_sizer = wx.BoxSizer(wx.VERTICAL)
+        global_sizer = wx.GridBagSizer(vgap=self.BORDER, hgap=self.BORDER)
+        global_sizer.SetEmptyCellSize((self.BORDER, self.BORDER))
 
-        sizer = wx.BoxSizer(wx.HORIZONTAL)
-        sizer.Add(self.status_button,
-                  flag=wx.ALL | wx.ALIGN_CENTER_VERTICAL,
-                  border=self.BORDER)
+        box = wx.StaticBox(self.root, label=self.HEADER_SETTINGS)
+        sizer = wx.StaticBoxSizer(box, wx.HORIZONTAL)
         sizer.Add(self.configure_button,
-                  flag=wx.TOP | wx.BOTTOM | wx.RIGHT | wx.ALIGN_CENTER_VERTICAL,
+                  1,
+                  flag=wx.ALL | wx.ALIGN_CENTER,
                   border=self.BORDER)
         sizer.Add(self.about_button,
-                  flag=wx.TOP | wx.BOTTOM | wx.RIGHT | wx.ALIGN_CENTER_VERTICAL,
+                  1,
+                  flag=wx.ALL | wx.ALIGN_CENTER,
                   border=self.BORDER)
-        global_sizer.Add(sizer)
-
-        sizer = wx.BoxSizer(wx.HORIZONTAL)
+        global_sizer.Add(sizer,
+                         flag=wx.EXPAND,
+                         pos=(0, 0),
+                         span=(1, 2))
+
+        box = wx.StaticBox(self.root, label=self.HEADER_OUTPUT)
+        self.status_sizer = wx.StaticBoxSizer(box, wx.HORIZONTAL)
+
+        self.status_sizer.Add(self.status_label,
+                              1,
+                              flag=wx.ALL | wx.ALIGN_CENTER, border=self.BORDER)
+        self.status_sizer.Add(self.status_button,
+                              1,
+                              flag=wx.ALL | wx.ALIGN_CENTER, border=self.BORDER)
+        global_sizer.Add(self.status_sizer,
+                         flag=wx.EXPAND,
+                         pos=(1, 0),
+                         span=(1, 2))
+
+        box = wx.StaticBox(self.root, label=self.HEADER_MACHINE)
+        sizer = wx.StaticBoxSizer(box, wx.HORIZONTAL)
         sizer.Add(self.spinner,
-                  flag=(wx.LEFT | wx.BOTTOM | wx.RIGHT | 
-                        wx.ALIGN_CENTER_VERTICAL), 
+                  flag=(wx.LEFT | wx.RIGHT |
+                        wx.ALIGN_CENTER_VERTICAL),
                   border=self.BORDER)
         sizer.Add(self.connection_ctrl, 
-                  flag=(wx.LEFT | wx.BOTTOM | wx.RIGHT | 
-                        wx.ALIGN_CENTER_VERTICAL), 
+                  flag=(wx.LEFT | wx.RIGHT |
+                        wx.ALIGN_CENTER_VERTICAL),
                   border=self.BORDER)
         longest_machine = max(machine_registry.get_all_names(), key=len)
-        longest_state = max((STATE_ERROR, STATE_INITIALIZING, STATE_RUNNING), 
+        longest_state = max((STATE_ERROR, STATE_INITIALIZING, STATE_RUNNING),
                             key=len)
         longest_status = '%s: %s' % (longest_machine, longest_state)
```

Comment:
Could we stick with `state` for machine state by making this `longest_state`? There's a terminology mismatch here between `state` vs `status`.

We also already have a `longest_status` in scope, which made me wonder why we were recomputing it initially.